WalletWasabi/Wallets/FileSystemBlockRepository.cs

```diff
@@ -15,17 +16,124 @@ namespace WalletWasabi.Wallets
 	/// </summary>
 	public class FileSystemBlockRepository : IRepository<uint256, Block>
 	{
-		public FileSystemBlockRepository(string blocksFolderPath, Network network)
+		public FileSystemBlockRepository(string blocksFolderPath, Network network, long targetBlocksFolderSizeMb = 300)
 		{
 			BlocksFolderPath = blocksFolderPath;
 			Network = network;
 			CreateFolders();
+			EnsureBackwardsCompatibility();
+			Prune(targetBlocksFolderSizeMb);
 		}
 
 		public string BlocksFolderPath { get; }
 		public Network Network { get; }
 		private AsyncLock BlockFolderLock { get; } = new AsyncLock();
 
+		private void EnsureBackwardsCompatibility()
+		{
+			try
+			{
+				// Before Wasabi 1.1.13
+				var wrongBlockFolderPath = Path.Combine(EnvironmentHelpers.GetDataDir(Path.Combine("WalletWasabi", "Client")), $"Blocks{Network}");
+				if (Directory.Exists(wrongBlockFolderPath))
+				{
+					using (BenchmarkLogger.Measure())
+					{
+						var cntSuccess = 0;
+						var cntRedundant = 0;
+						var cntFailure = 0;
+						foreach (var filePath in Directory.EnumerateFiles(wrongBlockFolderPath))
+						{
+							try
+							{
+								var fileName = Path.GetFileName(filePath);
+								var migrationPath = Path.Combine(BlocksFolderPath, fileName);
+
+								// Unintuitively File.Move overwrite: false throws an IOException if the file already exists.
+								// https://docs.microsoft.com/en-us/dotnet/api/system.io.file.move?view=netcore-3.1
+								if (!File.Exists(migrationPath))
+								{
+									File.Move(filePath, migrationPath, overwrite: false);
```

Comment:
Do we have magic move at our library?

Reply:
Maybe you have always miused File.Move like me. Microsoft is clear about it and I verified it.

Reply:
By replying to your other similar question, I realized I misunderstood this question of yours. So, I just checked it, no we don't have magic move in the IoHelpers. We have in the IoSafeDigestableMutex thingy, but none of those features are beneficial here.